host/sgx/enter.c

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <openenclave/internal/calls.h>
+#include <openenclave/internal/sgxtypes.h>
+#include "asmdefs.h"
+#include "enclave.h"
+
+// Define a variable with given name and bind it to the register with the
+// corresponding name. This allows manipulating the register as a normal
+// C variable. The variable and hence the register is also assigned the
+// specified value.
+#define OE_DEFINE_REGISTER(regname, value) \
+    register uint64_t regname __asm__(#regname) = (uint64_t)(value)
+
+#if _WIN32
+
+// In x64 Windows ABI, the frame pointer can be any register and the frame
+// pointer points to a constant location *within* the frame. In x64, the
+// frame pointer points to the top of the frame. Windows debugger extensions
+// for Open Enclave SDK require a linux-style frame pointer for the oe_enter
+// function on the host-side.
+#define OE_DEFINE_FRAME_POINTER(r, v) OE_DEFINE_REGISTER(r, v)
+
+// As per Windows x64 ABI, the linux style frame pointer is -0x40 bytes
+// from the address of the enclave parameter which is passed via the stack.
+// Enclave parameter is the 7th parameter. Including the return-address, the
+// Linux style frame-pointer is -(7+1)*8 = -64 = -0x40 bytes from the enclave
+// parameter in the stack.
+#define OE_FRAME_POINTER_VALUE ((uint64_t)&enclave - 0x40)
+#define OE_FRAME_POINTER , "r"(rbp)
+
+// The SDK currently does not use a bridge for stack sitching.
```

Comment:
Currently, Windows debuggers detect an ocall by the presence of a function named `__oe_dispatch_ocall` in the call stack.
Unlike oegdb ,they don't require that the Ocall stack be stitched, stitch the stack themselves. This might change in the future as we are trying to evolve the debugger contract to support both Linux and Windows.